src/v/iceberg/rest_client/catalog_client.cc

```diff
@@ -397,23 +479,29 @@ ss::future<expected<commit_table_response>> catalog_client::commit_table_update(
 }
 
 path_components::path_components(
-  std::optional<base_path> base,
-  std::optional<prefix_path> prefix,
-  std::optional<api_version> api_version)
+  std::optional<base_path> base, std::optional<api_version> api_version)
   : _base{trim_slashes(std::move(base), "")}
-  , _prefix{trim_slashes(std::move(prefix), "")}
   , _api_version{trim_slashes(std::move(api_version), "v1")} {}
 
+void path_components::reset_prefix(std::optional<prefix_path> path) {
+    _prefix = trim_slashes<prefix_path>(std::move(path), "");
+    vlog(
+      log.info, "Reset Iceberg REST catalog client prefix to '{}'", *_prefix);
```

Comment:
oh maybe this is what i was looking for

Reply:
Right exactly, we're just logging at info level once when the configuration happens.